tests/utils.py

```diff
@@ -649,7 +649,7 @@ def run(self, result=None):
         for test_name in dir(self):
             if test_name.startswith('test_'):
                 expect_failure = hasattr(self, 'not_implemented') and test_name in self.not_implemented
-                getattr(self, test_name).__dict__['__unittest_expecting_failure__'] = expect_failure
+                getattr(self, test_name).__dict__.setdefault('__unittest_expecting_failure__', expect_failure)
```

Comment:
The `getattr` is already being done on line 651 (actually: `hasattr` and using it).

Anyway, I see Travis-CI also reports a failure.

I'm preparing a pull-request which does the refactoring I was thinking about, but first running the tests locally to make sure no unexpected failures pop up.